tests/tracer/test_sampler.py

```diff
@@ -261,6 +261,31 @@ def test_sampling_rule_init():
     assert rule.name == name_regex
 
 
+def test_sampling_rule_init_via_env():
+    # Testing single sampling rule
+    with override_env(dict(DD_TRACE_SAMPLING_RULES='[{"sample_rate":1.0,"service":"xyz","name":"abc"}]')):
```

Comment:
we need examples for malformed input: either not valid json, extra field names, incorrect field names, incorrect value types (e.g. "1.0" instead of 1.0, etc).

we also need examples without sample_rate/service/name being set (all combinations), etc

Reply:
I'll add more scenarios into these tests

Reply:
In scenarios where the sampling rate isn't set or an invalid option is the expected behavior to drop/ignore that rule?
``` 
# Testing for only Sample rate greater than 1.0
    with override_env(dict(DD_TRACE_SAMPLING_RULES='[{"sample_rate":2.0,"service":"xyz","name":"abc"}]')):
        sampling_rule = DatadogSampler().rules
        assert sampling_rule == []
```